### PR TITLE
Thread pool fixes

### PR DESCRIPTION
**Pull request by johnsonjh**
* Thread pool fixes

### CODE REVIEW

File: tests/tests.c [hunk 1]
```diff
@@ -2084,12 +2084,19 @@ enum {
 };
 
 static bool threadpool_pseudojob(void* arg) {
-    sir_debug("this is a pseudo job that actually does nothing (arg: %p)", arg);
+    char thread_name[SIR_MAXPID];
```
Comment (aremmell) [inline tests/tests.c:2087]: = {0};

**Review by aremmell — DISMISSED** [review]

File: tests/tests.c (hunk 1 above)
Comment (johnsonjh) [reply tests/tests.c:2087]: k

**Review by johnsonjh — COMMENTED** [review]
OK!

**Comment by johnsonjh** [comment]
Let's see why CI doesn't like it.

File: src/sirthreadpool.c [hunk 2]
```diff
@@ -36,53 +36,74 @@ static unsigned __stdcall thread_pool_proc(void* arg);
...
+            thrd_err    = errno;
+            thrd_create = false;
+            break;
         }
 #endif
     }
 
+#if !defined(__WIN__)
+    op = pthread_attr_destroy(&attr);
+    SIR_UNUSED(op);
```
Comment (aremmell) [inline src/sirthreadpool.c]: SIR_ASSERT_UNUSED(0 == op, op);

File: tests/tests.c [hunk 3]
```diff
@@ -2084,12 +2084,19 @@ enum {
...
-    sir_debug("this is a pseudo job that actually does nothing (arg: %p)", arg);
+    char thread_name[SIR_MAXPID] = {0};
+
+    _sir_snprintf_trunc(thread_name, SIR_MAXPID, "pool.%p", arg);
+    (void)_sir_setthreadname(thread_name);
+
+    (void)sir_debug("start of pseudo job that does nothing (arg: %p)", arg);
+
 #if !defined(__WIN__)
     sleep(1);
```
Comment (aremmell) [inline tests/tests.c]: We can replace this with `sir_sleep_msec(1000)` now

**Review by aremmell — APPROVED** [review]

**Comment by johnsonjh** [comment]
**`file-archive-large`** is the failing test on CI on these.

**Review by aremmell — APPROVED** [review]

**Pull request merged**